test/e2e/plain_provisioner_test.go

```diff
@@ -594,6 +594,85 @@ var _ = Describe("plain provisioner bundle", func() {
 				}).Should(BeNil())
 			})
 		})
+
+		When("the bundle is backed by a private repository", func() {
+			var (
+				bundle      *rukpakv1alpha1.Bundle
+				secret      *corev1.Secret
+				privaterepo string
+			)
+			BeforeEach(func() {
+				privaterepo = os.Getenv("PRIVATE_REPO")
+				username := os.Getenv("PRIVATE_REPO_USERNAME")
+				password := os.Getenv("PRIVATE_REPO_PASSWORD")
+				secret = &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "gitsecret-",
+						Namespace:    defaultSystemNamespace,
+					},
+					Data: map[string][]byte{"username": []byte(username), "password": []byte(password)},
+					Type: "Opaque",
+				}
+				err := c.Create(ctx, secret)
+				Expect(err).To(BeNil())
+				bundle = &rukpakv1alpha1.Bundle{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "combo-git-branch",
+					},
+					Spec: rukpakv1alpha1.BundleSpec{
+						ProvisionerClassName: plain.ProvisionerID,
+						Source: rukpakv1alpha1.BundleSource{
+							Type: rukpakv1alpha1.SourceTypeGit,
+							Git: &rukpakv1alpha1.GitSource{
+								Repository: "https://github.com/" + privaterepo,
+								Ref: rukpakv1alpha1.GitRef{
+									Branch: "main",
+								},
+								Auth: rukpakv1alpha1.Authorization{
+									Secret: corev1.LocalObjectReference{
+										Name: secret.Name,
+									},
+								},
+							},
+						},
+					},
+				}
+				err = c.Create(ctx, bundle)
+				Expect(err).To(BeNil())
+			})
+
+			AfterEach(func() {
+				err := c.Delete(ctx, bundle)
+				Expect(err).To(BeNil())
+				err = c.Delete(ctx, secret)
+				Expect(err).To(BeNil())
+			})
+
+			It("Can create and unpack the bundle successfully", func() {
+				if privaterepo == "" {
```

Comment:
IMO this should probably fail earlier, since the test can't succeed if the repo isn't set